vach/src/loader/archive.rs

```diff
@@ -54,18 +47,34 @@ impl<T: Seek + Read> Archive<T> {
 		Header::validate(&header, config)?;
 
 		// Generate and store Registry Entries
+		let mut use_decryption = false;
 		let mut entries = HashMap::new();
 		for _ in 0..header.capacity {
 			let (entry, id) =
-				RegistryEntry::from_handle(&mut handle, header.flags.contains(Flags::SIGNED_FLAG))?;
+				RegistryEntry::from_handle(&mut handle)?;
+			if entry.flags.contains(Flags::ENCRYPTED_FLAG) && !use_decryption {
+				use_decryption = true;
+			};
 			entries.insert(id, entry);
+		};
+
+		// Build decryptor
+		let mut decryptor = None;
+
+		if use_decryption {
+			if let Some(pk) = config.public_key {
+				decryptor = Some(EDCryptor::new(&pk, config.magic))
+			} else {
+				println!("WARNING! Some leafs require decryption, yet no public key was provided")
```

Comment:
We should probably use log for this, so users can opt to ignore this crate's logs.

Reply:
Logging and error types are coming, just keep this unnecessary `println!` pinned